runtime/nvqir/CircuitSimulator.h

```diff
@@ -596,7 +598,7 @@ class CircuitSimulatorBase : public CircuitSimulator {
 
   /// @brief Add the given number of qubits to the state.
   virtual void addQubitsToState(std::size_t count,
-                                const std::complex<double> *state = nullptr) {
+                                const void *state = nullptr) {
```

Comment:
```suggestion
                                 void *state = nullptr) {
```
Can we use parametric polymorphism as a type adaptor instead of a bare pointer here? (Something like a `std::variant`.)